src/factory.ts

```diff
@@ -37,22 +40,91 @@ function runCommand(
 
 function isGitHubRelease(
   command: string,
-  options: GitHubReleaseOptions | ReleasePROptions
-): options is GitHubReleaseOptions {
+  options: GitHubReleaseFactoryOptions | ReleasePRFactoryOptions
+): options is GitHubReleaseFactoryOptions {
   return command === 'github-release' && typeof options === 'object';
 }
 
 function run(runnable: ReleasePR | GitHubRelease) {
   return runnable.run();
 }
 
-function githubRelease(options: GitHubReleaseOptions): GitHubRelease {
-  return new GitHubRelease(options);
+function githubRelease(options: GitHubReleaseFactoryOptions): GitHubRelease {
+  const {
+    repoUrl,
+    defaultBranch,
+    token,
+    apiUrl,
+    octokitAPIs,
+    releaseType,
+    path,
+    packageName,
+    bumpMinorPreMajor,
+    releaseAs,
+    snapshot,
+    monorepoTags,
+    fork,
+    changelogSections,
+    lastPackageVersion,
+    versionFile,
+    ...remaining
+  } = options;
+
+  const github = gitHubInstance({
+    repoUrl,
+    defaultBranch,
+    token,
+    apiUrl,
+    octokitAPIs,
+  });
+  const prClass = releaseType ? releasePRClass(releaseType) : ReleasePR;
+  const releasePR = new prClass({
+    github,
+    path,
+    packageName,
+    bumpMinorPreMajor,
+    releaseAs,
+    snapshot,
+    monorepoTags,
+    fork,
+    changelogSections,
+    lastPackageVersion,
+    versionFile,
+  });
+
+  return new GitHubRelease({github, releasePR, ...remaining});
```

Comment:
I like this approach, this ends up meaning that the GitHub release can use any releaser specific methods on `releasePR`?

Reply:
`GitHubRelease` [already relies on a `ReleasePR`](https://github.com/googleapis/release-please/blob/b277b8909e78047797796c0428f545ce482a0b2a/src/github-release.ts#L85) instance for to perform it's one job (`run`) - this change is simple removing the knowledge of how to construct it (and thus also removing any duplicate constructor options) and centralizing it in case other future classes (aka aggregate) need one too. Same pattern as pulling out the building of a `GitHub` instance because that was already being duplicated (by `ReleasePR`)